sdks/python/apache_beam/transforms/core.py

```diff
@@ -1714,14 +1713,20 @@ def MapTuple(fn, *args, **kwargs):  # pylint: disable=invalid-name
 
   # Proxy the type-hint information from the original function to this new
   # wrapped function.
-  type_hints = get_type_hints(fn)
+  type_hints = get_type_hints(fn).with_defaults(
+      typehints.decorators.IOTypeHints.from_callable(fn))
   if type_hints.input_types is not None:
-    wrapper = with_input_types(
-        *type_hints.input_types[0], **type_hints.input_types[1])(
-            wrapper)
+    # FIXME(feyu): ignore input hints, as we do not have enough
```

Comment:
Is this tied to a different BEAM issue as a future fix?

Reply:
Not yet. I'll file an issue and update the fixme tag. A fix is likely non-trivial, because it seems like our data model is insufficient (as the comment noted)